engine/src/flutter/lib/web_ui/test/engine/text_editing_test.dart

```diff
@@ -1943,6 +1943,126 @@ Future<void> testMain() async {
       hideKeyboard();
     });
 
+    test('Supports composition changes with shift arrow selection in Japanese IME', () {
+      final HybridTextEditing testTextEditing = HybridTextEditing();
+      final GlobalTextEditingStrategySpy editingStrategy = GlobalTextEditingStrategySpy(
+        testTextEditing,
+      );
+
+      showKeyboard(inputType: 'text');
+      editingStrategy.composingText = 'へんかん';
+      sendFrameworkMessage(
+        codec.encodeMethodCall(
+          const MethodCall('TextInput.setEditingState', <String, dynamic>{
+            'text': 'へんかん',
+            'selectionBase': 4,
+            'selectionExtent': 4,
+            'composingBase': 0,
+            'composingExtent': 4,
```

Comment:
Does Flutter always send `composingBase` and `composingExtent` when the user is composing text? If yes, then we don't really need the new `isComposing` boolean, right? We can simply check for the presence of `composingBase` and/or `composingExtent` inside `EditingState.applyToDomElement`.